src/main/msp/msp_serial.c

```diff
@@ -526,6 +549,11 @@ void mspSerialProcess(mspEvaluateNonMspData_e evaluateNonMspData, mspProcessComm
                 const uint8_t c = serialRead(mspPort->port);
                 const bool consumed = mspSerialProcessReceivedData(mspPort, c);
 
+                if (mspPort->c_state == MSP_CLI_CMD) {
```

Comment:
This flow control is terrible. Existing code structure is bad enough and this is even worse.

Reply:
Adjusted